Color-Game-using-Tkinter/script.py

```diff
@@ -0,0 +1,150 @@
+# Author: PriyanshuChatterjee
+# Colour Game using Tkinter
+# See the readme.md for how to play and which file to run to play.
+
+# importing the modules
+import tkinter
+import random
+
+# list of possible colours
+colours = ['Red', 'Blue', 'Green', 'Pink', 'Black',
+           'Yellow', 'Orange', 'White', 'Purple', 'Brown']
+
+# function to start the game`
+def startGame(event):
+    if timeleft == 30:
+        #starting the countdown timer.
+        countdown()
+
+    # calling the function to choose the next colour
+    nextColour()
+
+# Function to choose and display the next colour
+def nextColour():
+    # Using the globally declared variable 'score' and 'timeLeft' variables
+    # declared above
+    global score
+    global timeleft
+
+    # if a game is currently in session
+    if timeleft > 0:
+
+        # making the box to write the names of the colour active
+        e.focus_set()
+
+        # if the colour typed and colour of the text same
+        if e.get().lower() == colours[1].lower():
+            score += 1
+
+        # clearing the text entry box
+        e.delete(0, tkinter.END)
+
+        random.shuffle(colours)
+
+        # changing the colour to type, by changing the text and the colour
+        # to a random colour
+        label.config(fg=str(colours[1]), text=str(colours[0]))
+
+        # updating the score
+        scoreLabel.config(text="Score: " + str(score))
+
+# Declaring and defining the countdown timer function
+def countdown():
+    global timeleft
+
+    # if a game is in session
+    if timeleft > 0:
+        # decrement the timer:
+        timeleft -= 1
+
+        # updating the time left Label
+        timeLabel.config(text="Time left: "
+                              + str(timeleft))
+
+        # calling the function again after 1 second
+        timeLabel.after(1000, countdown)
+
+# The Game GUI starting Function
+def vp_start_gui():
+    global root
+    # Score of the player, initially zero
+    global score
```

Comment:
Define the score variable outside the functions, that's why test failed

Reply:
I cannot implement it by defining them outside the functions.